### PR TITLE
[Docs] Add EBNF to sampling params docs 

### PR DESCRIPTION
**Pull request by adarshxs**
## Modifications

updates sampling_params.md and openai_api_completions.ipynb with ebnf

### CODE REVIEW

**Comment by merrymercy** [comment]
Fix lint? https://sgl-project.github.io/references/contributor_guide.html#format-your-code

**Comment by zhaochenyang20** [comment]
@adarshxs Hello, whould you like to update your branch to fix the CI?

It's pretty easy: `pre-commit run --show-diff-on-failure --color=always --all-files`

**Comment by adarshxs** [comment]
@zhaochenyang20 apologies for the linting error 😅 fixed it

File: docs/references/sampling_params.md [hunk 1]
```diff
@@ -58,13 +58,18 @@ ignore_eos: bool = False,
 skip_special_tokens: bool = True,
 # Whether to add spaces between special tokens during detokenization.
 spaces_between_special_tokens: bool = True,
-# Constrains the output to follow a given regular expression.
-regex: Optional[str] = None,
 # Do parallel sampling and return `n` outputs.
 n: int = 1,
+
+## Structured Decoding Constraints
```
Comment (merrymercy) [inline docs/references/sampling_params.md]: Call it `Structured Outputs` in all places

File: docs/references/sampling_params.md [hunk 2]
```diff
@@ -179,25 +184,36 @@ print(response.json())
 The `image_data` can be a file name, a URL, or a base64 encoded string. See also `python/sglang/srt/utils.py:load_image`.
 Streaming is supported in a similar manner as [above](#streaming).
 
-### Structured decoding (JSON, Regex)
-You can specify a JSON schema or a regular expression to constrain the model output. The model output will be guaranteed to follow the given constraints.
+### Structured decoding (JSON, Regex, EBNF)
+You can specify a JSON schema, Regular Expression or [EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) to constrain the model output. The model output will be guaranteed to follow the given constraints.
```
Comment (merrymercy) [inline docs/references/sampling_params.md:188]: mention xgrammar uses GGML BNF format currently and add a link to the specification https://github.com/ggerganov/llama.cpp/blob/master/grammars/README.md

File: docs/references/sampling_params.md [hunk 3]
```diff
@@ -179,25 +184,36 @@ print(response.json())
...
+
+SGLang supports two grammar backends:
+
+- [Outlines](https://github.com/dottxt-ai/outlines) (default): Supports JSON schema and Regular Expression constraints.
+- [XGrammar](https://github.com/mlc-ai/xgrammar): Supports JSON schema and EBNF constraints.
+
+> 🔔 Only one constraint parameter (`json_schema`, `regex`, or `ebnf`) can be specified at a time.
+
+Initialise grammar backend using `--grammar-backend xgrammar` flag
```
Comment (merrymercy) [inline docs/references/sampling_params.md]:
```suggestion
Initialise xgrammar backend using `--grammar-backend xgrammar` flag
```

File: docs/backend/openai_api_completions.ipynb [hunk 4]
```diff
@@ -220,10 +220,16 @@
...
    "metadata": {},
    "source": [
-    "## Structured decoding (JSON, Regex)\n",
-    "You can define a JSON schema or regular expression to constrain the model's output. The model output will be guaranteed to follow the given constraints and this depends on the grammar backend.\n",
+    "## Structured decoding (JSON, Regex, EBNF)\n",
+    "You can define a JSON schema, regular expression or EBNF grammar to constrain the model's output. The model output will be guaranteed to follow the given constraints and this depends on the grammar backend.\n",
     "\n",
-    "SGlang has two backends: [Outlines](https://github.com/dottxt-ai/outlines) (default) and [XGrammar](https://blog.mlc.ai/2024/11/22/achieving-efficient-flexible-portable-structured-generation-with-xgrammar). Xgrammar accelerates JSON decoding performance but does not support regular expressions. To use Xgrammar, add the `--grammar-backend xgrammar` when launching the server:\n",
+    "SGLang has two grammar backends: \n",
+    "- Outlines (default) - `json_schema` & `regex`\n",
```
Comment (merrymercy) [inline docs/backend/openai_api_completions.ipynb]: We should add back the links. Can you copy the whole introduction from `docs/references/sampling_params.md:L187`?

**Review by merrymercy — CHANGES_REQUESTED** [review]
This looks good! I left some comments.

File: docs/references/sampling_params.md [hunk 5]
```diff
@@ -179,25 +184,37 @@ print(response.json())
 The `image_data` can be a file name, a URL, or a base64 encoded string. See also `python/sglang/srt/utils.py:load_image`.
 Streaming is supported in a similar manner as [above](#streaming).
 
-### Structured decoding (JSON, Regex)
-You can specify a JSON schema or a regular expression to constrain the model output. The model output will be guaranteed to follow the given constraints.
+### Structured Outputs (JSON, Regex, EBNF)
+You can specify a JSON schema, Regular Expression or [EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) to constrain the model output. The model output will be guaranteed to follow the given constraints. XGrammar uses
```
Comment (merrymercy) [inline docs/references/sampling_params.md]: The tailing "XGrammar uses " is redundant?

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: docs/references/sampling_params.md (hunk 5 above)
Comment (adarshxs) [reply docs/references/sampling_params.md]: yes removing it and fixing the linting

**Comment by adarshxs** [comment]
@merrymercy done please have a look

**Comment by merrymercy** [comment]
@adarshxs Thanks. It is merged.

**Pull request merged**